core/src/impl/Kokkos_ViewCtor.hpp

```diff
@@ -199,11 +203,22 @@ struct ViewCtorProp : public ViewCtorProp<void, P>... {
       Kokkos::Impl::has_type<AllowPadding_t, P...>::value;
   static constexpr bool initialize =
       !Kokkos::Impl::has_type<WithoutInitializing_t, P...>::value;
+  static constexpr bool host_serial_init =
+      Kokkos::Impl::has_type<HostSerialInit_t, P...>::value;
+  static_assert(initialize || !host_serial_init,
+                "Incompatible WithoutInitializing and HostSerialInit view "
+                "alloc properties");
 
   using memory_space    = typename var_memory_space::type;
   using execution_space = typename var_execution_space::type;
   using pointer_type    = typename var_pointer::type;
 
+  // FIXME can't quite do that to validate user in[ut because we sometimes
+  // attach default instances ourselves
+  // static_assert(!host_serial_init || !has_execution_space,
+  //              "Incompatible HostSerialInit view "
+  //              "alloc property with execution space argument");
+
```

Comment:
@dalg24 - we don't need this PR if you want to drop it. Only have one instance so far in empire that wasn't using the phalanx object that is already fixed.

Reply:
Rewriting Phalanx' VoV abstraction in terms of this might not be the worse idea.
In any case there seems to be a need for this feature.  Turns out the misuse of VoVs is more widespread than we suspected.

Reply:
Was the "I am tempted to withdraw this change." comment just referring to the commented out static_assert here, not the whole feature then?

Reply:
The FIXME comment with the assertion